oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/PropertyValueImpl.java

```diff
@@ -125,12 +125,7 @@ public void restrict(FilterImpl f, Operator operator, PropertyValue v) {
                     f.restrictPath(v.getValue(Type.STRING), PathRestriction.EXACT);
                 }
             } else {
-                if (operator == Operator.NOT_EQUAL && v != null) {
-                    // "x <> 1" also means "x is not null"
-                    f.restrictProperty(pn, Operator.NOT_EQUAL, null, propertyType);
```

Comment:
If we keep this restriction (in addition to the f.restrictProperty(pn, operator, v, propertyType)), then property indexes can continue to use it.

Reply:
@thomasmueller  - propertyIndex could still use it - because in restrictProperty method in FilterImpl, I added new varaibles to handle inequality, which would effectively leave the old behaviour for PropertyIndex as it is. 

Changes in Filter.java and FilterImpl.java should handle this.

btw, there's a case with older version of lucene that fails - (version 1) - I am still looking into that , so there might be further changes here.

Reply:
Oh..I see.
>And to me it seems there are not enough test cases in the property index, maybe.
Yes, because no tests are failing with current changes.

Also, I tried to add both the property restrictions i.e
f.restrictProperty(pn, Operator.NOT_EQUAL, null, propertyType);
and 
f.restrictProperty(pn, operator, v, propertyType));

But it seems that the FilterImpl only returns the first one when we try and get all the PropertyRestrictions for a given property, so only the first of the 2 restrictions are honored.

Reply:
@thomasmueller  - thanks. I understand now.

>And to me it seems there are not enough test cases in the property index, maybe.

Yes, all the tests were passing and I was unaware of this usage in ContentMirrorStoreStrategy. 

I have made some changes in the latest commit - 
https://github.com/apache/jackrabbit-oak/pull/465/commits/f9d45d8f8f422cdf9f27f1aca794ae73e9b5e0bc. Now adding 2 propertyRestrictions - so that the old behaviour with PropertyIndex is not broken.

Can you please re review ?